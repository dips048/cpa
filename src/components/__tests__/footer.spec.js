import React from "react"
import renderer from "react-test-renderer"
import Footer from "../footer"
import { render } from "@testing-library/react"

describe("Footer", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<Footer companyName={"MyCompany"} />)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("should render footer cards `Ask a Question`, `We Are Here to Help` & `Send Us a File`" , () => {
    const { getByText, getAllByTestId, getAllByRole } = render(<Footer companyName={"MyCompany"} />)
    expect(getByText("Send Us a File")).toBeInTheDocument()
    expect(getByText("We Are Here to Help")).toBeInTheDocument()
    expect(getByText("Ask a Question")).toBeInTheDocument()
    expect(getAllByTestId("fa-icon").length).toEqual(4)
    expect(getAllByRole("button").length).toEqual(4)
  })
})
