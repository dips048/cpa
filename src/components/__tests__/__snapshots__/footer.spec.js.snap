// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer renders correctly 1`] = `
<footer>
  <div
    className="row text-center mx-auto container"
  >
    <div
      className="col-lg-4 col-md-6 my-3 mx-auto text-center"
    >
      <div
        className="icon-color"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-comment fa-5x "
          data-icon="comment"
          data-prefix="far"
          data-testid="fa-icon"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M256 32C114.6 32 .0272 125.1 .0272 240c0 47.63 19.91 91.25 52.91 126.2c-14.88 39.5-45.87 72.88-46.37 73.25c-6.625 7-8.375 17.25-4.625 26C5.818 474.2 14.38 480 24 480c61.5 0 109.1-25.75 139.1-46.25C191.1 442.8 223.3 448 256 448c141.4 0 255.1-93.13 255.1-208S397.4 32 256 32zM256.1 400c-26.75 0-53.12-4.125-78.38-12.12l-22.75-7.125l-19.5 13.75c-14.25 10.12-33.88 21.38-57.5 29c7.375-12.12 14.37-25.75 19.88-40.25l10.62-28l-20.62-21.87C69.82 314.1 48.07 282.2 48.07 240c0-88.25 93.25-160 208-160s208 71.75 208 160S370.8 400 256.1 400z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <h3>
        Ask a Question
      </h3>
      <p>
        Find comfort in knowing an Expert in accounting is only an email or phone-call away.
      </p>
      <button
        className="btn btn-primary btn-lg"
      >
        <a
          href="tel:(800) 896-4500"
        >
          (800) 896-4500
        </a>
      </button>
    </div>
    <div
      className="col-lg-4 col-md-6 my-3 mx-auto text-center"
    >
      <div
        className="icon-color"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-calendar fa-5x "
          data-icon="calendar"
          data-prefix="far"
          data-testid="fa-icon"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 448 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M152 64H296V24C296 10.75 306.7 0 320 0C333.3 0 344 10.75 344 24V64H384C419.3 64 448 92.65 448 128V448C448 483.3 419.3 512 384 512H64C28.65 512 0 483.3 0 448V128C0 92.65 28.65 64 64 64H104V24C104 10.75 114.7 0 128 0C141.3 0 152 10.75 152 24V64zM48 448C48 456.8 55.16 464 64 464H384C392.8 464 400 456.8 400 448V192H48V448z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <h3>
        We Are Here to Help
      </h3>
      <p>
        We will happily offer you a free consultation to determine how we can best serve you.
      </p>
      <button
        className="btn btn-primary btn-lg"
      >
        <a
          href="/contact"
        >
          Contact Us
        </a>
      </button>
    </div>
    <div
      className="col-lg-4 col-md-6 my-3 mx-auto text-center"
    >
      <div
        className="icon-color"
      >
        <svg
          aria-hidden="true"
          className="svg-inline--fa fa-copy fa-5x "
          data-icon="copy"
          data-prefix="far"
          data-testid="fa-icon"
          focusable="false"
          role="img"
          style={Object {}}
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M502.6 70.63l-61.25-61.25C435.4 3.371 427.2 0 418.7 0H255.1c-35.35 0-64 28.66-64 64l.0195 256C192 355.4 220.7 384 256 384h192c35.2 0 64-28.8 64-64V93.25C512 84.77 508.6 76.63 502.6 70.63zM464 320c0 8.836-7.164 16-16 16H255.1c-8.838 0-16-7.164-16-16L239.1 64.13c0-8.836 7.164-16 16-16h128L384 96c0 17.67 14.33 32 32 32h47.1V320zM272 448c0 8.836-7.164 16-16 16H63.1c-8.838 0-16-7.164-16-16L47.98 192.1c0-8.836 7.164-16 16-16H160V128H63.99c-35.35 0-64 28.65-64 64l.0098 256C.002 483.3 28.66 512 64 512h192c35.2 0 64-28.8 64-64v-32h-47.1L272 448z"
            fill="currentColor"
            style={Object {}}
          />
        </svg>
      </div>
      <h3>
        Send Us a File
      </h3>
      <p>
        Use our convenient SecureSend page to securely deliver a file directly to a member of our firm.
      </p>
      <button
        className="btn btn-primary btn-lg"
      >
        <a
          href="/resources/secure-send"
        >
          Secure Send
        </a>
      </button>
    </div>
  </div>
  <div
    className="container"
  >
    <h2>
      Newsletter
    </h2>
    <p>
      Subscribe to our newsletter to receive news, updates, and valuable tips.
    </p>
    <div
      className="mx-auto"
    >
      <form
        className="col-sm-8 mx-auto d-inline"
      >
        <input
          name="email"
          placeholder="Email"
          type="text"
        />
        <button
          aria-label="submit"
          className="btn btn-primary"
          name="submit"
          type="submit"
        >
          <svg
            aria-hidden="true"
            className="svg-inline--fa fa-rss "
            data-icon="rss"
            data-prefix="fas"
            data-testid="fa-icon"
            focusable="false"
            role="img"
            style={Object {}}
            viewBox="0 0 448 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M25.57 176.1C12.41 175.4 .9117 185.2 .0523 198.4s9.173 24.65 22.39 25.5c120.1 7.875 225.7 112.7 233.6 233.6C256.9 470.3 267.4 480 279.1 480c.5313 0 1.062-.0313 1.594-.0625c13.22-.8438 23.25-12.28 22.39-25.5C294.6 310.3 169.7 185.4 25.57 176.1zM32 32C14.33 32 0 46.31 0 64s14.33 32 32 32c194.1 0 352 157.9 352 352c0 17.69 14.33 32 32 32s32-14.31 32-32C448 218.6 261.4 32 32 32zM63.1 351.9C28.63 351.9 0 380.6 0 416s28.63 64 63.1 64s64.08-28.62 64.08-64S99.37 351.9 63.1 351.9z"
              fill="currentColor"
              style={Object {}}
            />
          </svg>
        </button>
      </form>
    </div>
  </div>
  <div
    className="footer-nav"
  >
    <div
      className="container"
    >
      <div
        className="linkConatainer"
      >
        <ul
          className="navLinks"
        >
          <li>
            <h3>
              Contact
            </h3>
          </li>
          <li>
            MyCompany
          </li>
          <li>
            <a
              href="tel:(800) 896-4500"
            >
              (800) 896-4500
            </a>
          </li>
          <li>
            <a
              href="mailto:support@cpasitesolutions.com"
            >
              support@cpasitesolutions.com
            </a>
          </li>
        </ul>
      </div>
      <hr />
      <div
        className="linkConatainer"
      >
        <ul
          className="linkConatainerB"
        >
          <li
            className="ps-2"
          >
            <span>
              Â© 
              MyCompany
               2022
            </span>
          </li>
          <li
            className="ps-2"
          >
            <a
              href="?"
            >
              Site Map
            </a>
          </li>
          <li
            className="ps-2"
          >
            <a
              href="?"
            >
              Privacy Policy
            </a>
          </li>
          <li
            className="ps-2"
          >
            <a
              href="?"
            >
              Disclaimer
            </a>
          </li>
        </ul>
        <div
          className="text-center"
        >
          <a
            className="m-1"
            href="https://www.facebook.com"
            title="Facebook"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-facebook "
              data-icon="facebook"
              data-prefix="fab"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M504 256C504 119 393 8 256 8S8 119 8 256c0 123.8 90.69 226.4 209.3 245V327.7h-63V256h63v-54.64c0-62.15 37-96.48 93.67-96.48 27.14 0 55.52 4.84 55.52 4.84v61h-31.28c-30.8 0-40.41 19.12-40.41 38.73V256h68.78l-11 71.69h-57.78V501C413.3 482.4 504 379.8 504 256z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
          <a
            className="m-1"
            href="https://www.twitter.com"
            title="Twitter"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-twitter "
              data-icon="twitter"
              data-prefix="fab"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M459.4 151.7c.325 4.548 .325 9.097 .325 13.65 0 138.7-105.6 298.6-298.6 298.6-59.45 0-114.7-17.22-161.1-47.11 8.447 .974 16.57 1.299 25.34 1.299 49.06 0 94.21-16.57 130.3-44.83-46.13-.975-84.79-31.19-98.11-72.77 6.498 .974 12.99 1.624 19.82 1.624 9.421 0 18.84-1.3 27.61-3.573-48.08-9.747-84.14-51.98-84.14-102.1v-1.299c13.97 7.797 30.21 12.67 47.43 13.32-28.26-18.84-46.78-51.01-46.78-87.39 0-19.49 5.197-37.36 14.29-52.95 51.65 63.67 129.3 105.3 216.4 109.8-1.624-7.797-2.599-15.92-2.599-24.04 0-57.83 46.78-104.9 104.9-104.9 30.21 0 57.5 12.67 76.67 33.14 23.72-4.548 46.46-13.32 66.6-25.34-7.798 24.37-24.37 44.83-46.13 57.83 21.12-2.273 41.58-8.122 60.43-16.24-14.29 20.79-32.16 39.31-52.63 54.25z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
          <a
            className="m-1"
            href="https://www.linkedin.com"
            title="LinkedIn"
          >
            <svg
              aria-hidden="true"
              className="svg-inline--fa fa-linkedin "
              data-icon="linkedin"
              data-prefix="fab"
              focusable="false"
              role="img"
              style={Object {}}
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M416 32H31.9C14.3 32 0 46.5 0 64.3v383.4C0 465.5 14.3 480 31.9 480H416c17.6 0 32-14.5 32-32.3V64.3c0-17.8-14.4-32.3-32-32.3zM135.4 416H69V202.2h66.5V416zm-33.2-243c-21.3 0-38.5-17.3-38.5-38.5S80.9 96 102.2 96c21.2 0 38.5 17.3 38.5 38.5 0 21.3-17.2 38.5-38.5 38.5zm282.1 243h-66.4V312c0-24.8-.5-56.7-34.5-56.7-34.6 0-39.9 27-39.9 54.9V416h-66.4V202.2h63.7v29.2h.9c8.9-16.8 30.6-34.5 62.9-34.5 67.2 0 79.7 44.3 79.7 101.9V416z"
                fill="currentColor"
                style={Object {}}
              />
            </svg>
          </a>
        </div>
      </div>
    </div>
  </div>
</footer>
`;
